api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionEditPaths.scala

```
@@ -92,38 +94,68 @@ trait SubmissionEditPaths
+  def justifyMacroEditPath(institutionId: String)(implicit ec: ExecutionContext) =
+    path("filings" / Segment / "submissions" / IntNumber / "edits" / "macro") { (period, seqNr) =>
+      timedPost { uri =>
+        entity(as[MacroEditJustificationWithName]) { justifyEdit =>
+          completeVerified(institutionId, period, seqNr, uri) {
+            val fValidator = fHmdaFileValidator(SubmissionId(institutionId, period, seqNr))
+
+            val fValidationState = for {
+              v <- fValidator
+              j <- (v ? JustifyMacroEdit(justifyEdit.edit, justifyEdit.justification)).mapTo[MacroEditJustified]
+              state <- getValidationState(institutionId, period, seqNr)
+            } yield state
+            completeValidationState("macro", fValidationState, uri, "")
+          }
+        }
+      }
+    }
+
+  // institutions/<institutionId>/filings/<period>/submissions/<seqNr>/edits/quality
+  def verifyQualityEditsPath(institutionId: String)(implicit ec: ExecutionContext) =
+    path("filings" / Segment / "submissions" / IntNumber / "edits" / "quality") { (period, seqNr) =>
+      timedPost { uri =>
+        entity(as[QualityEditsVerification]) { verification =>
+          completeVerified(institutionId, period, seqNr, uri) {
+            val verified = verification.verified
+            val fSubmissionsActor = (supervisor ? FindSubmissions(SubmissionPersistence.name, institutionId, period)).mapTo[ActorRef]
+            val subId = SubmissionId(institutionId, period, seqNr)
+            val fValidator = fHmdaFileValidator(subId)
+
+            val fSubmissions = for {
+              va <- fValidator
+              v <- va ? VerifyQualityEdits(verified)
+              sa <- fSubmissionsActor
+              s <- (sa ? GetSubmissionById(subId)).mapTo[Submission]
+            } yield s
+
+            onComplete(fSubmissions) {
+              case Success(submission) =>
```

I updated the `for` comprehension so it maps the return message from `VerifyQualityEdits` to its expected type. In my tests, this catches the case where the call fails or returns the wrong type.

(I could be wrong--dealing with futures is complex--but) I don't think it's possible for this to fail silently. If line 132 never returns, the `for` comprehension just won't yield anything.